test/skate_web/controllers/auth_controller_test.exs

```diff
@@ -43,6 +43,36 @@ defmodule SkateWeb.AuthControllerTest do
       assert Guardian.Plug.current_claims(conn)["groups"] == ["test1", "skate-readonly"]
     end
 
+    test "redirects to the post_auth_return_to path if available", %{conn: conn} do
+      mock_auth = %Ueberauth.Auth{
+        provider: :keycloak,
+        uid: "test_username",
+        credentials: %Ueberauth.Auth.Credentials{
+          expires_at: System.system_time(:second) + 1_000,
+          refresh_token: "test_refresh_token"
+        },
+        info: %{email: "test@mbta.com"},
+        extra: %Ueberauth.Auth.Extra{
+          raw_info: %UeberauthOidcc.RawInfo{
+            userinfo: %{
+              "resource_access" => %{
+                "test-client" => %{"roles" => ["test1", "skate-readonly"]}
+              }
+            }
+          }
+        }
+      }
+
+      conn =
+        conn
+        |> assign(:ueberauth_auth, mock_auth)
+        |> init_test_session(%{post_auth_return_to: ~p"/detours"})
+        |> get(~p"/auth/keycloak/callback")
```

Comment:
suggestion(non-blocking): I wonder if there's a way to test this without "mocking" the redirect path, and instead attempt to navigate to `/detours`, get redirected, go though login, and then assert we end up at `/detours`?

Reply:
Follow-up task: https://app.asana.com/0/1152340551558956/1207199406216195/f

I assigned it to myself so that it stays on my radar, but I'm not attached to _me_ being the one to do it.